### DISCUSSION TITLE
Feature Request: Once the add-on download from the Add-on Store is finished, a sound notification should be triggered to alert the user.

### DISCUSSION THREAD

**Discussion by cary-rowen** [nvaccess/nvda#15117] [Ideas]
### Is your feature request related to a problem? Please describe.
Upon the completion of the add-on download, users currently lack an immediate notification. To ascertain the status of the download, they need to follow these possible steps:

1. Shift the system focus back and forth between the downloaded  item and an adjacent item, after which NVDA will eventually announce the download status.
2. Utilize table navigation to reach the rightmost column.
3. Press the 'Esc' key; NVDA will then alert you if the download has not been completed.

All the aforementioned methods are excessively complex, particularly when multiple add-ons are being downloaded simultaneously.
### Describe the solution you'd like
Provide an audible notification once the task has finished downloading. This sound alert can be kept brief to minimize disruption. I am even able to supply some sample sounds for reference.
### Describe alternatives you've considered
Preserve the current status.
### Additional context
None

**Comment by vgjh2005** [comment]
Hi:
I had made two audio files. I will upload here.
Thanks.
[audio.zip](https://github.com/nvaccess/nvda/files/11925607/audio.zip)

**Comment by XLTechie** [comment]
@vgjh2005 Some thoughts about the proposed sound samples:

I like the sounds used in the first one.

Could you try the same synth setup, but go from the lower to the high, so it 
resolves on the major up chord? I.E. this, but the other way? Not just reversed, 
because then the attacks are wrong, but re-order the notes.


The second, however, sounds too much like the old Windows 3.11through XP error 
sound to my ears--the one that was called "exclamation.wav" It's also a little 
on the quiet side.

**Comment by seanbudd** [comment]
Is it important that the user knows when the downloads are finished?
These are background tasks that shouldn't effect the user.
When exiting the dialog we block the exit on installation - we could also block the dialog exit by waiting for downloads to finish with an intermediate progress dialog, rather than encouraging the user to cancel the downloads or return to the dialog

**Comment by cary-rowen** [comment]
If the user does not know whether the add-on has been downloaded, there are only two situations after the user clicks ESC:
1. The add-on has finished downloading and NVDA asks the user to restart to complete the add-on installation.
2. The add-on has not finished downloading, and NVDA asks the user to cancel the download task.

If the user expects the installation to be performed as soon as the download task completes, they have no way of knowing if the download is complete.

**Comment by seanbudd** [comment]
>If the user expects the installation to be performed as soon as the download task completes, they have no way of knowing if the download is complete.

Does knowing if the download is complete matter if it's just a background task followed as part of installation?

>The add-on has not finished downloading, and NVDA asks the user to cancel the download task.

My earlier proposal is to replace this with waiting until the download and pre-installation completes, or ask the user to return to the store.

**Comment by hwf1324** [comment]
> Does knowing if the download is complete matter if it's just a background task followed as part of installation?

Yes, I think it's important. The user may need to use his downloaded add-on immediately, at which point he can restart using the add-on immediately if he is notified that the download is complete.

**Comment by seanbudd** [comment]
>The user may need to use his downloaded add-on immediately, at which point he can restart using the add-on immediately if he is notified that the download is complete.

This is not how the installation workflow works. An add-on is downloaded, then pre-install tasks are run. After these pre-install tasks are finished, a user can restart NVDA to complete the installation.
The pre-install tasks are not run until the user exits the add-on store. This means when exiting the add-on store dialog, you must always wait until the pre-installation is complete before restarting NVDA.
This means that if a user is alerted when the download is complete, they are still blocked on the pre-install action when exiting the add-on store.
As such, getting an alert that the download has completed means you still have to wait for the pre-install tasks, so you can't take any action based on the sound notification, you are still blocked on the pre-installation.
To me, this means the alert is pointless, as you still cannot take any actions or act differently based on the notification.

**Comment by cary-rowen** [comment]
Hi,

Sorry for being late, I don't think this should be turned into a discussion, it's still a valid feature request.
@seanbudd 
You may have misunderstood me.
The core of this request has always been that the user wants to know when the download task is complete.
This will take a long time if users have downloaded multiple add-ons at the same time, or downloaded a large add-on, however they want to manually restart NVDA immediately after the download task is complete to complete the add-on installation.
Currently, to do this, the user has to manually check the download status, as described in my original description.
Based on the above, I think it might be better to have a progress bar while the add-on is downloading than to play a sound when the task download is complete.
If you get my point and want me to reopen a feature request, or want me to revise this request, please let me know.

Thanks

**Reply by seanbudd** [reply]
unfortunately discussions cannot be converted back to issues.
Could you open a new feature request for a progress bar while add-ons download when exiting the dialog.
